### PR TITLE
fix: [BB-326] fields and soil styling fixes

### PR DESCRIPTION
**Pull request by kcaparas1630**
# Description

Fixed Fields and soil styling fixes to be almost the same as mockups.

## Type of changes

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have already been accepted and merged

## Further comments

---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-sustainment-capstone-2024-332-frontend.apps.silver.devops.gov.bc.ca)
- [Backend](https://nr-sustainment-capstone-2024-332-backend.apps.silver.devops.gov.bc.ca/api/health)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/merge.yml)

### CODE REVIEW

File: frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsListComponent.tsx [hunk 1]
```diff
@@ -34,89 +32,120 @@ const FieldsListComponent: FC<FieldListProps> = ({ farmDetails }) => {
...
-                  </StyledListItem>
-                  <StyledListItem width="180px">
-                    <div className="smallItems">
-                      <h2>pH</h2>
-                      <p>{HasSoilTest ? SoilTest.valPH : highPH}</p>
-                    </div>
-                  </StyledListItem>
-                </StyledListItemGroup>
-              </NutrientsFieldListGroup>
+              {/* <NutrientsFieldListGroup> */}
```
Comment (GDamaso) [inline frontend/src/Commons/Forms/InputModules/FieldsAndSoil/FieldsListComponent.tsx:91]: Should this stay here?

File: frontend/src/Commons/Forms/InputModules/ListComponent.styles.ts [hunk 2]
```diff
@@ -6,27 +6,41 @@
...
+`;
+const StyledListContainer = styled.div<StyledListType>`
   position: relative;
   display: flex;
-  flex-direction: column;
+  flex-direction: row;
   padding-top: 20px;
-  gap: ${(props) => (props.formCrops ? '50px' : '0')};
+  gap: ${(props) => props.gap};
+  align-items: flex-start; // Add this line
```
Comment (GDamaso) [inline frontend/src/Commons/Forms/InputModules/ListComponent.styles.ts:39]: It sure was added :laughing:

File: frontend/src/Commons/Forms/InputModules/ListComponent.styles.ts [hunk 3]
```diff
@@ -6,27 +6,41 @@
...
 
   @media (min-width: ${screenSizes.desktop}) {
-    flex-direction: row;
+    // gap: 120px;
```
Comment (GDamaso) [inline frontend/src/Commons/Forms/InputModules/ListComponent.styles.ts:42]: :eyes:

File: frontend/src/Commons/Forms/InputModules/ListComponent.styles.ts [hunk 4]
```diff
@@ -53,28 +67,27 @@ const StyledListItem = styled.div<StyledListType>`
...
   padding-top: 8px;
-  position: absolute;
-  right: 0;
+  margin-left: auto;
 
   @media (min-width: ${screenSizes.desktop}) {
     font-size: 32px;
     padding-top: 20px;
+    // position: absolute;
+    // right: 0;
```
Comment (GDamaso) [inline frontend/src/Commons/Forms/InputModules/ListComponent.styles.ts:90]: :eyes:

File: frontend/src/Constants/ComponentText.ts [hunk 5]
```diff
@@ -2,7 +2,7 @@ const ComponentText = {
   NEXT: 'Next',
   ADD: 'Add',
   CANCEL: 'Cancel',
-  ADD_FIELD: 'Add Field',
+  ADD_FIELD: 'New Field',
```
Comment (GDamaso) [inline frontend/src/Constants/ComponentText.ts:5]: Good!

File: frontend/src/Commons/Button/Button.style.ts [hunk 6]
```diff
@@ -31,7 +33,7 @@ const StyledButton = styled.button<StyledButtonProps>`
   border-radius: ${(props) => props.radius};
   border: ${(props) =>
     props.actions === 'primary' ? 0 : `1px solid ${tokens.surfaceColorBorderMedium}`};
-  padding: ${(props) => (props.landingPageButton ? '20px' : '0')};
+  padding: ${(props) => (props.landingPageButton ? '20px' : '')};
```
Comment (GDamaso) [inline frontend/src/Commons/Button/Button.style.ts:36]: Do you mean to leave it undefined?

File: frontend/src/Commons/Button/FieldButtonGroup.styles.ts [hunk 7]
```diff
@@ -52,7 +52,7 @@ const StyledNewFieldButtonContainer = styled.div<FormProps>`
   position: relative;
   top: 15px;
   width: 100%;
-  margin-bottom: ${(props) => (props.formCrops ? '40cpx' : '0')};
+  margin-bottom: ${(props) => (props.formCrops ? '40px' : '0')};
```
Comment (GDamaso) [inline frontend/src/Commons/Button/FieldButtonGroup.styles.ts:55]: I bet it took forever to find this damn typo :smile: 
They are the worst...

File: frontend/src/Commons/CustomLink/CustomLink.style.ts [hunk 8]
```diff
@@ -14,15 +14,15 @@ const StyledLinkContainer = styled.div<StyledLinkProps>`
   height: 100%;
   width: 100%;
   max-height: 42px;
-  max-width: ${(props) => getButtonSize(props.size, false)};
+  max-width: ${(props) => getButtonSize(props.size, false, false, false)};
```
Comment (GDamaso) [inline frontend/src/Commons/CustomLink/CustomLink.style.ts:17]: What's up here?

File: frontend/src/Commons/FormStyles.styles.ts [hunk 9]
```diff
@@ -95,12 +97,13 @@ const StyledButtonGroupContainer = styled.div<FormProps>`
 const StyledTestContainer = styled.div`
   display: flex;
   flex-direction: column;
-  gap: 10px;
+  gap: 11px;
```
Comment (GDamaso) [inline frontend/src/Commons/FormStyles.styles.ts:100]: One small pixel between divs,
One huge leap for the gaps

**Review by GDamaso — COMMENTED** [review]
I think there's still room for improvement. The New Field text is overflowing inside the button for me. The alignment between the Next button from Farm Information and Fields and Soil are different:

image

New Field Button alignment is also off on mobile:
image

Please review and test it and let me know when you want another review.

File: frontend/src/Commons/CustomLink/CustomLink.style.ts (hunk 8 above)
Comment (kcaparas1630) [reply frontend/src/Commons/CustomLink/CustomLink.style.ts:17]: added params on getButtonSize method util

**Comment by GDamaso** [comment]
I like the alignment of the Field and Sampling month, but could it be aligned with Area and sampling month instead? There's a gap there that could be removed this way.

**Review by GDamaso — APPROVED** [review]
Good job bud, let's go! :rocket:

**Pull request merged**